### PR TITLE
boards/same54-xpro: support for board variation with SST26VF064B flash

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

New variants of the `same54-xpro` board now come with the SST26VF064B QSPI flash instead of N25Q256A.
This requires some changes:

 - the capacity encoding scheme is different from existing chips
 - Global Block-Protection Unlock is required before the chip accepts writes (but this is only needed once)
 - 8 MiB device instead of 32 MiB, does *only* support 24 bit addressing

To solve the third problem, `addr_width` is not determined at run-time so it does not have to be set by the board anymore.

### Testing procedure

I used `tests/mtd_raw` to test the changes:

#### old same54-xpro

```
2022-02-03 22:03:44,478 # main(): This is RIOT! (Version: 2022.04-devel-223-g9fd1b-drivers/mtd_spi_nor-microchip)
2022-02-03 22:03:44,479 # Manual MTD test
2022-02-03 22:03:44,483 # init MTD_0… OK (32 MiB)
2022-02-03 22:03:44,486 # init MTD_1… OK (256 byte)
> test 0
2022-02-03 22:03:47,197 # test 0
2022-02-03 22:03:47,197 # [START]
2022-02-03 22:03:48,471 # [SUCCESS]
```

#### new same54-xpro

```
> 2022-02-03 22:03:19,956 # main(): This is RIOT! (Version: 2022.04-devel-223-g9fd1b-drivers/mtd_spi_nor-microchip)
2022-02-03 22:03:19,958 # Manual MTD test
2022-02-03 22:03:19,961 # init MTD_0… OK (8 MiB)
2022-02-03 22:03:19,964 # init MTD_1… OK (256 byte)
> test 0
2022-02-03 22:03:24,709 # test 0
2022-02-03 22:03:24,710 # [START]
2022-02-03 22:03:25,978 # [SUCCESS]
```

### CODE REVIEW

**Review by dylad — APPROVED** [review]
Looks good to me.
I don't have the hardware to test but I trust @benpicco testing.
Let's see what CI thinks about it.

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 1]
```diff
@@ -303,6 +306,24 @@ static uint32_t mtd_spi_nor_get_size(const mtd_jedec_id_t *id)
         (id->device[1] & ~0x3) == 0) {
         return (0x1F & id->device[0]) * MBIT_AS_BYTES;
     }
+    if (mtd_spi_manuf_match(id, SPI_NOR_JEDEC_MICROCHIP)) {
+        switch (id->device[1]) {
+        case 0x12:  /* SST26VF020A */
+        case 0x8c:  /* SST25VF020B */
+            return  2 * MBIT_AS_BYTES;
```
Comment (maribu) [inline drivers/mtd_spi_nor/mtd_spi_nor.c:313]:
```suggestion
            return 2 * MBIT_AS_BYTES;
```

The CI doesn't like indent after the `return`.

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 2]
```diff
@@ -303,6 +306,24 @@ static uint32_t mtd_spi_nor_get_size(const mtd_jedec_id_t *id)
...
+        case 0x54:  /* SST26WF040B */
+        case 0x8d:  /* SST25VF040B */
+            return  4 * MBIT_AS_BYTES;
```
Comment (maribu) [inline drivers/mtd_spi_nor/mtd_spi_nor.c:316]:
```suggestion
            return 4 * MBIT_AS_BYTES;
```

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 3]
```diff
@@ -303,6 +306,24 @@ static uint32_t mtd_spi_nor_get_size(const mtd_jedec_id_t *id)
...
+        case 0x58:  /* SST26WF080B */
+        case 0x8e:  /* SST25VF080B */
+            return  8 * MBIT_AS_BYTES;
```
Comment (maribu) [inline drivers/mtd_spi_nor/mtd_spi_nor.c:319]:
```suggestion
            return 8 * MBIT_AS_BYTES;
```

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 4]
```diff
@@ -303,6 +306,24 @@ static uint32_t mtd_spi_nor_get_size(const mtd_jedec_id_t *id)
...
+        case 0x42:  /* SST26VF032B */
+            return 32 * MBIT_AS_BYTES;
+        case 0x43:  /* SST26VF064B */
+        case 0x53:  /* SST26WF064C */
+            return 64 * MBIT_AS_BYTES;
+        }
+    }
 
     /* everyone else seems to use device ID 2 for density */
     return 1 << id->device[1];
```
Comment (maribu) [inline drivers/mtd_spi_nor/mtd_spi_nor.c:329]: Yeah, the Microchip approach indeed looks so much more elegant! Why should obtaining the flash size should be implemented in an easy, ROM efficient and future proof way when we can have a fun look up table instead?

**Review by maribu — APPROVED** [review]
Code looks good to me and you provided testing for both 24 and 32 bit address widths flash chips.

**Pull request merged**